compiler/codegen/symbol.cpp

```diff
@@ -2227,19 +2256,52 @@ GenRet FnSymbol::codegenPYXType() {
   if (retType != dtVoid) {
     if (retType == exportTypeChplBytesWrapper) {
 
-      // The raw result of the routine call, a "chpl_bytes_wrapper".
-      funcCall += "cdef chpl_bytes_wrapper rdat = ";
+      // The raw result of the routine call, a "chpl_byte_buffer".
+      funcCall += "cdef chpl_byte_buffer rv = ";
+
+      // Unpack the required fields.
+      returnStmt += "\tcdef char* rdata = rv.data\n";
+      returnStmt += "\tcdef Py_ssize_t rsize = rv.size\n";
+
+      // Create a new Python bytes that is a copy of the Chapel bytes.
+      returnStmt += "\tret = PyBytes_FromStringAndSize(rdata, rsize)\n";
+
+      // This will free the "chpl_byte_buffer" buffer if required.
+      returnStmt += "\tchpl_byte_buffer_free(rv)\n";
+
+      Type* origt = getUnwrappedRetType(this)->getValType();
+      INT_ASSERT(origt != NULL);
+
+      // The only two types that can be mapped to "chpl_byte_buffer".
+      INT_ASSERT(origt == dtBytes || origt == dtString);
+
+      //
+      // If the original Chapel routine's return type is a string, then
+      // decode the Python bytes into a Python UTF-8 string.
+      //
+      if (origt == dtString) {
+        returnStmt += "\tret = ret.decode(\'utf-8\')\n";  
+      }
+
+    } else if (retType == dtStringC) {
+
+      //
+      // NOTE: The result of the routine call is a `c_string`, AKA const
+      // char*. We know that the only thing we could be returning here
+      // is a `c_string` (since Chapel strings are returned as
+      // "chpl_byte_buffer"), which means that we can't call `chpl_free` on
+      // it because we know nothing about its lifetime.
+      //
+      funcCall += "cdef const char* rdata = <char*> ";
 
-      // Convert the "chpl_bytes_wrapper" struct to a Python bytes.
-      returnStmt += "\tcdef char* rdata = rdat.data\n";
-      returnStmt += "\tcdef Py_ssize_t rsize = rdat.size\n";
-      returnStmt += "\tcdef char* v = rdat.data\n";
+      // Get the length of the "char*" using strlen.
+      returnStmt += "\tcdef Py_ssize_t rsize = strlen(rdata)\n";
 
       // Create a new Python bytes that is a copy of the Chapel string.
       returnStmt += "\tret = PyBytes_FromStringAndSize(rdata, rsize)\n";
 
-      // This will free the "chpl_bytes_wrapper" buffer if required.
-      returnStmt += "\tchpl_bytes_wrapper_free(rdat)\n";
+      // Decode the bytes into a Python string.
```

Comment:
Did we decide to convert c_strings into Python strings as well?  That seems inconsistent with Cython's normal behavior towards c_strings.

Reply:
The question I was asking myself is what we want Chapel to map the `c_string` to in Python interop (rather than what Cython does, as I believe I can brute force my way around that). It seems reasonable to me to map C strings to Python strings.

I'd be happy to revert if we want to map to bytes.

For instance, I didn't consider existing code that might break due to this change (again 🙄).

Reply:
I think it'd be good to follow the Cython behavior, since 1) c_string should go away eventually, and 2) it's really intended to be "just a C string" so decisions Cython made about C's strings should be respected

Reply:
Ok, I'll continue to convert `c_string` to bytes.